pkg/reconciliation/construct_service_test.go

```diff
@@ -407,3 +408,99 @@ func TestAddingAdditionalLabels(t *testing.T) {
 		t.Errorf("service labels = %v, want %v", service.Labels, expected)
 	}
 }
+
+func TestServicePorts(t *testing.T) {
+	tests := []struct {
+		name                string
+		dc                  *api.CassandraDatacenter
+		dcServicePorts      []int32
+		allPodsServicePorts []int32
+	}{
+		{
+			name: "Cassandra 3.11.14",
+			dc: &api.CassandraDatacenter{
+				Spec: api.CassandraDatacenterSpec{
+					ClusterName:   "bob",
+					ServerType:    "cassandra",
+					ServerVersion: "3.11.14",
+				},
+			},
+			dcServicePorts:      []int32{8080, 9000, 9042, 9103, 9142, 9160},
+			allPodsServicePorts: []int32{8080, 9000, 9042, 9103},
```

Comment:
I wonder: shouldn't all-pods service expose the same ports as the dc service? But I didn't want to change that in this PR.

Reply:
it depends what the all-ports service is used for.
If no one expressed any requirement around that, I agree we shouldn't change it.

Reply:
cass-operator sets allPods as the ServiceName for all pods to ensure that they get DNS working correctly. Using only the dcPods would make the DNS names fail in certain cases:

https://github.com/k8ssandra/cass-operator/pull/18